includes/data-stores/class-wc-admin-reports-downloads-data-store.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * WC_Admin_Reports_Downloads_Data_Store class file.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Reports_Downloads_Data_Store.
+ */
+class WC_Admin_Reports_Downloads_Data_Store extends WC_Admin_Reports_Data_Store implements WC_Admin_Reports_Data_Store_Interface {
+
+	/**
+	 * Table used to get the data.
+	 *
+	 * @var string
+	 */
+	const TABLE_NAME = 'wc_download_log';
+
+	/**
+	 * Mapping columns to data type to return correct response types.
+	 *
+	 * @var array
+	 */
+	protected $column_types = array(
+		'id'          => 'intval',
+		'date'        => 'strval',
+		'date_gmt'    => 'strval',
+		'download_id' => 'strval', // String because this can sometimes be a hash.
+		'file_name'   => 'strval',
+		'product_id'  => 'intval',
+		'order_id'    => 'intval',
+		'user_id'     => 'intval',
+		'ip_address'  => 'strval',
+	);
+
+	/**
+	 * SQL columns to select in the db query and their mapping to SQL code.
+	 *
+	 * @var array
+	 */
+	protected $report_columns = array(
+		'id'          => 'download_log_id as id',
+		'date'        => 'timestamp as date_gmt',
+		'download_id' => 'product_permissions.download_id',
+		'product_id'  => 'product_permissions.product_id',
+		'order_id'    => 'product_permissions.order_id',
+		'user_id'     => 'product_permissions.user_id',
+		'ip_address'  => 'user_ip_address as ip_address',
+	);
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		global $wpdb;
+	}
+
+	/**
+	 * Updates the database query with parameters used for downloads report.
+	 *
+	 * @param array $query_args Query arguments supplied by the user.
+	 * @return array            Array of parameters used for SQL query.
+	 */
+	protected function get_sql_query_params( $query_args ) {
```

Comment:
Whew some pretty intensive query setup here, very legible and easy to follow though, nice work.